pkg/controller/nodedaemon/reconcile.go

```diff
@@ -41,6 +48,24 @@ type DaemonReconciler struct {
 func (r *DaemonReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	r.reqLogger = logf.Log.WithName(controllerName).WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
 
+	// do a one-time delete of the old static-provisioner daemonset
+	if !r.deletedStaticProvisioner {
+		provisioner := &appsv1.DaemonSet{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: ProvisionerName, Namespace: request.Namespace}, provisioner)
+		if err == nil { // provisioner daemonset found
+			r.reqLogger.Info(fmt.Sprintf("old daemonset %q found, cleaning up", ProvisionerName))
+			err = r.client.Delete(context.TODO(), provisioner)
+			if err != nil && !errors.IsGone(err) {
+				r.reqLogger.Error(err, fmt.Sprintf("could not delete daemonset %q", ProvisionerName))
+				return reconcile.Result{}, err
+			}
+		} else if !errors.IsNotFound(err) { // unknown error
+			r.reqLogger.Error(err, fmt.Sprintf("could not fetch daemonset %q to clean it up", ProvisionerName))
+			return reconcile.Result{}, err
+		}
+		r.deletedStaticProvisioner = true
+	}
```

Comment:
Should we wait for daemonset replicas to be 0 before reconciling the `localvolumeset`  further?

Reply:
Is there going to be a problem if they both run https://github.com/openshift/local-storage-operator/blob/master/deploy/scripts/quick_reset.sh at the same time? Even that seems unlikely unless the controller-manager is down or something.

Reply:
Can do though

Reply:
Yes that would be nice. I don't like the idea that we might have 2 ways of doing things running concurrently.

Reply:
There are at least couple of ways of ensuring that pods created by daemonset are torn down before we can start reconciling the `LocalVolumeSet` object further:

1. We could check for all pods created by daemonset to be deleted. I think it should be possible to do that by a selector etc.
2. There was another neat trick I saw on stackoverflow where one can set a non-existant node-selector on daemonset which would result in all pods to be deleted and then we can delete the daemonset.

I am sure there could be other ways of doing this.

Reply:
I added a wait to confirm the pods with the label app=localvolumeset-local-provisioner are gone before setting deletedLocalProvisioner